src/v/cluster/tm_stm.h

```diff
@@ -171,6 +171,11 @@ class tm_stm final : public persisted_stm {
 
     std::vector<kafka::transactional_id> get_expired_txs();
 
+    using get_txs_result = checked<
```

Comment:
a hash map is used as an index inside of tm_stm and since kafka::transactional_id is already a member of tm_transaction returning a map doest convey any more information to the caller but creates an unnecessary coupling - imagine that we want to change the index and use another data structure: we either would need to change the consumer of `get_all_transactions` too  or create a proxy to a map when it isn't really need a map.

can you please change signature of get_all_transactions and return vector?

Reply:
the difference between killed and aborted is very subtle lets convert killed into aborted and also filter out tombstones (in admin_server)

Reply:
@rystsov what volume of transactions might we expect? tm_transaction is a pretty good size structure. if may be a lot, consider using our internal `fragmented_vector` which doesn't store everything contiguously and can avoid memory fragmentation issues for large vectors.

Reply:
it's up to the users, we keep all ongoing transactions in memory so it may be a lot, `fragmented_vector` sounds good too, I just was concerned about exposing `tm_stm` internals

Reply:
I can not use fragmented_vector, because fragment size should be power of 2, and % sizeof(T) == 0. But sizeof(tm_transaction) = 120 bytes. I can increase this size to power of 2, but prefer to use vector instead of fragmented_vector.

Reply:
@dotnwat I don't think we should expect users to touch admin api often so using `vector` instead of `fragmented_vector` may be fine, what do you think?